designer/client/components/ComponentCreate/ComponentCreate.tsx

```diff
@@ -21,6 +21,7 @@ import { ComponentContext } from "../../reducers/component/componentReducer";
 import "./ComponentCreate.scss";
 
 function useComponentCreate(props) {
+  const [renderTypeEdit, setRenderTypeEdit] = useState<boolean>(false);
```

Comment:
no need to change, just an fyi - you don't need to add the type `<boolean>` if you provide a default value. the overload(s) for use state are using generics. It is useful/helpful in terms of visibility for more complex types though 

```
const [a, setA] = useState(true ? "a" : "b")
setA(new Date()) 
//<html>TS2345: Argument of type 'Date' is not assignable to parameter of type 'SetStateAction<string>'.<br/>Type 'Date' is not assignable to type 'string'.
```

Reply:
force of habit :)